gremlin-client/src/aio/connection.rs

```diff
@@ -283,10 +284,24 @@ fn receiver_loop(
                 }
                 Some(Ok(item)) => match item {
                     Message::Binary(data) => {
-                        let response: Response = serde_json::from_slice(&data).unwrap();
+                        let response = deserializer
+                            .read_response(data)
+                            .expect("Unable to parse message");
                         let mut guard = requests.lock().await;
+
+                        //GraphBinary permits a null response request id, so in lieu of a request id assume
+                        //a single entry in the requests to be the one we should respond to given connection
+                        //multiplexing isn't currently implemented
```

Comment:
At least on the GraphBinary side its documented that request Id can come back null. This seemed like the best means to try to mitigate that when it happens. Since connection multiplexing isn't in play, it seems like this should be an okay tradeoff.